File: CPP/Program/Car.cpp
/*Saya Legi Kuswandi mengerjakan LP4 dalam mata kuliah DPBO
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang 
telah dispesifikasikan. Aamiin.*/

// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
#include <iostream>
#include <string>
#include "Vehicle.cpp"

// Deklarasi namespace
using namespace std;

class Car : public Vehicle
{
private:
    string jumlah_kursi;
    string jumlah_pintu;
public:
    // Constructor
    Car()
    {

    }

    // Constructor : Overloading
    Car(string jumlah_kursi, string platnomor, string merk, string tahun_produksi, string warna) : Vehicle(platnomor, merk, tahun_produksi, warna){
        this->jumlah_kursi = jumlah_kursi;
        jumlah_pintu = "";
    }

    Car(string jumlah_kursi, string jumlah_pintu, string platnomor, string merk, string tahun_produksi, string warna) : Vehicle(platnomor, merk, tahun_produksi, warna){
        this->jumlah_kursi = jumlah_kursi;
        this->jumlah_pintu = jumlah_pintu;
    }

    // Method setter untuk set nilai setiap atribut pada kelas Car
    void setJumlah_kursi(string jumlah_kursi){
        this->jumlah_kursi = jumlah_kursi;
    }

    // Method getter untuk mendapatkan nilai setiap atribut pada kelas Car
    string getJumlahkursi(){
        return jumlah_kursi;
    }

    // Method setter untuk set nilai setiap atribut pada kelas Car
    void setJumlah_pintu(string jumlah_pintu){
        this->jumlah_pintu = jumlah_pintu;
    }

    // Method getter untuk mendapatkan nilai setiap atribut pada kelas Car
    string getJumlah_pintu(){
        return jumlah_pintu;
    }
    
    // Destructor
    ~Car()
    {

    }
};

File: Python/Program/Car.py
from Vehicle import Vehicle

class Car(Vehicle):
    def __init__(self, jumlah_kursi="", jumlah_pintu="", platnomor="", merk="", tahun_produksi="", warna=""):
        # Memanggil konstruktor kelas induk (Vehicle)
        super().__init__(platnomor, merk, tahun_produksi, warna)
        # Inisialisasi atribut khusus mobil
        self.jumlah_kursi = jumlah_kursi  # Jumlah kursi mobil
        self.jumlah_pintu = jumlah_pintu  # Jumlah pintu mobil

    # Setter untuk jumlah kursi mobil
    def set_jumlah_kursi(self, jumlah_kursi: str):
        self.jumlah_kursi = jumlah_kursi

    # Getter untuk jumlah kursi mobil
    def get_jumlah_kursi(self) -> str:
        return self.jumlah_kursi

    # Setter untuk jumlah pintu mobil
    def set_jumlah_pintu(self, jumlah_pintu: str):
        self.jumlah_pintu = jumlah_pintu

    # Getter untuk jumlah pintu mobil
    def get_jumlah_pintu(self) -> str:
        return self.jumlah_pintu
